test/test.js

```diff
@@ -1242,6 +1242,54 @@ describe("follow-redirects", function () {
           assert.equal(body.host, "localhost:3600");
         });
     });
+
+    it("uses the existing host header if redirect host is the same", function () {
```

Comment:
I expect this test already works with the existing code (but it's good to have it anyways).

Reply:
see comments below.